pawn-items.lic

```diff
@@ -0,0 +1,84 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#pawn-items
+=end
+
+custom_require.call %w[common common-items common-travel equipmanager]
+
+class PawnItems
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'source', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'Source container' },
+        { name: 'noun', regex: /^[A-z\.\s\-]+$/i, optional: true, variable: true, description: 'If specified, only items with this noun will be sold.' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+
+    # Put away items in hands to mitigate accidentally selling them
+    EquipmentManager.new.return_held_gear
+    # Warn the script user that this is designed to sell items off your person
+    warn_before_sell(args.source, args.noun)
+    # Turn items into coins
+    sell_items(args.source, args.noun)
+  end
+
+  def sell_items(source, noun)
+    # If container is very full then LOOK may not list all of them.
+    # If you're moving a specific item, then sort those to the top
+    # to increase chances we find and sell all of them in one go.
+    DRC.bput("sort #{noun} in my #{source}", "are now at the top", "What were you referring to", "Please rephrase that command", "You may only sort items in your inventory") if noun
+    DRCI.get_item_list(source, 'look')
+      .map { |full_name| full_name =~ /lot of other stuff/ ? full_name : full_name.split(' ').last }
+      .select { |item| noun ? /\b#{noun}\b/ =~ item : true }
+      .each do |item|
+        # This indicates there is more items than LOOK can show
+        # and we've reached the end of what we last saw.
+        # We need to look again to see what we can now see.
+        # Keep doing this until exhaust source.
+        if item =~ /lot of other stuff/
```

Comment:
Is there a chance we never exhaust the source? E.g. you use a noun and there are none in the source, but the source has a ton of non-noun items in it?

Reply:
Great question. I took a few minutes to walk through this some more to double check and I don't think this will go in to infinite recursion.

My conclusion is that because "lot of other stuff" will be the last item in the list, each prior item will follow the code path to line 44 to offload the contents.

If the item cannot be removed from the container then the script exits (avoids infinite recursion of rerummaging but never reducing item count).

If the item can be removed from the container then it's either sold (and so the container is becoming less full over time) or the item is returned to the container and the script exits (avoids infinite recursion again).

Is there a scenario you're thinking of that don't fall in to the above?

Reply:
> you use a noun and there are none in the source, but the source has a ton of non-noun items in it?

If you provided the `noun` argument, then lines 30 and 33 should ensure that if any items do have that noun then they'll be at the top of the container. Line 34 can only occur for items are the "noun" you're looking for.